src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    get_location_type
+)
+from knack.arguments import CLIArgumentType
+from argcomplete.completers import FilesCompleter
+from ._validators import scope_validator
+
+parameter_type = CLIArgumentType(
+    options_list=['--parameters'],
+    help='Parameters in JSON format or file path to Json file with "@" prefix.'
+)
+
+template_type = CLIArgumentType(
+    options_list=['--template'],
+    help='ARM template in JSON format or file path to Json file with "@" prefix.'
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('blueprint', validator=scope_validator) as c:
+        c.argument('management_group', help='The management group for the scope of the resource. The scope could either be a subscription or a management group. For blueprint assignments management group scope is reserved for future use. If management-group is not specified, will use subscription as scope')
+        c.ignore('scope')
+
+    with self.argument_context('blueprint create') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('display_name', help='One-liner string explain this resource.')
+        c.argument('description', help='Multi-line explain this resource.')
+        c.argument('target_scope', arg_type=get_enum_type(['subscription', 'managementGroup']), help='The scope where this blueprint definition can be assigned.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters required by this blueprint definition.')
+
+    with self.argument_context('blueprint import') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('input_path', help='The directory path for json definitions of blueprint and artifacts. Artifacts json files should be in input-path/artifacts/.', completer=FilesCompleter())
+
+    with self.argument_context('blueprint update') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('description', help='Multi-line explain this resource.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters required by this blueprint definition.')
+
+    with self.argument_context('blueprint delete') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint show') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint list') as c:
+        pass
+
+    with self.argument_context('blueprint artifact delete') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')

```

Comment:
Here we can give an options_list=['--name','-n'] for artifact name because it is the target resource now.

Reply:
updated